### PR TITLE
Doxygen marked files.  Forth subset.

### PR DESCRIPTION
**Pull request by MatthewMasarik-NOAA**
# Pull Request Summary
Forth subset of `doxygen` marked source files.

## Description
Provide a detailed description of what this PR does.
* 10 F90 source files have been marked-up with doxygen tags.  

What bug does it fix, or what feature does it add?
* This PR adds doxygen documentation.

Is a change of answers expected from this PR?
* No.  Only documentation is added.

Please also include the following information: 
* Add any suggestions for a reviewer 
  * N/A. 
 
* Mention any labels that should be added:  
  * _documentation_.
 
* Are answer changes expected from this PR? Please describe the changes and the reason why in addition to which of the following labels would apply.
  * No.  No active code was edited.  Only documentation was added by means of doxygen marked comments.

### Issue(s) addressed
* Please list any issues associated with this PR, including those the PR will fix/close. For example:  
- Addresses #586, though does not close.  The issue will be closed when all files in checklist have been marked off.  

### Commit Message
Doxygen marked source files, 4th subset.

### Check list  
- [x] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
- [x] Checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop). 
- [na] If a version number update is required, checked the [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) checklist. 
- [na] If a new feature was added, a regression test for testing the new feature is added. 
 
### Testing
* How were these changes tested?
  * The full matrix of regtests run against current develop.

* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)
  * The changes are not covered by regression tests as we do not test for documentation.

* Have the matrix regression tests been run (if yes, please note HPC and compiler)?
  * Yes.  Hera / Intel.

* Please indicate the expected changes in the regression test output, (Note the [list](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-develop#4-look-at-results) of known non-identical tests.)
  * The only expected changes are the known non-identical tests.

* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):
  * [matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/9551399/matrixDiff.txt)
  * [matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/9551400/matrixCompFull.txt)
  * [matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/9551401/matrixCompSummary.txt)

```bash
**********************************************************************                                                          
********************* non-identical cases ****************************                                                          
**********************************************************************                                                          
mww3_test_03/./work_PR2_UNO_MPI_d2                     (12 files differ)                                                        
mww3_test_03/./work_PR1_MPI_d2                     (19 files differ)                                                            
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (15 files differ)                                                      
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)                                                       
mww3_test_03/./work_PR3_UNO_MPI_d2                     (15 files differ)                                                        
mww3_test_03/./work_PR2_UQ_MPI_d2                     (15 files differ)                                                         
mww3_test_03/./work_PR3_UNO_MPI_e_c                     (1 files differ)                                                        
mww3_test_03/./work_PR3_UQ_MPI_d2                     (15 files differ)                                                         
ww3_ta1/./work_UPD0F_U                     (0 files differ)                                                                     
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)                                                                 
ww3_tp2.16/./work_MPI_OMPH                     (5 files differ)                                                                 
ww3_tp2.6/./work_ST0                     (1 files differ)                                                                       
ww3_tp2.6/./work_ST4                     (1 files differ)                                                                       
ww3_ufs1.3/./work_a                     (1 files differ)                                                                                                                                                                                    
**********************************************************************                                                          
************************ identical cases *****************************                                                          
**********************************************************************
```

### CODE REVIEW

File: model/src/w3idatmd.F90 [hunk 1]
```diff
@@ -257,6 +270,16 @@ MODULE W3IDATMD
 !/
       CONTAINS
 !/ ------------------------------------------------------------------- /
+!>
+!> @brief Set up the number of grids to be used.
+!>
+!> @details Use data stored in NGRIDS in W3GDATMD.
+!>
+!> @param[in] NDSE
```
Comment (JessicaMeixner-NOAA) [inline model/src/w3idatmd.F90]: The variable definitions are: 
!       NDSE    Int.   I   Error output unit number.
! NDST Int. I Test output unit number.
Can we add that here?

Comment (MatthewMasarik-NOAA) [reply model/src/w3idatmd.F90]: Yes!  Good catch.  I'll make that update.

Comment (MatthewMasarik-NOAA) [reply model/src/w3idatmd.F90]: Done.

**Comment by JessicaMeixner-NOAA** [comment]
[matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/9577660/matrixCompFull.txt)
[matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/9577661/matrixCompSummary.txt)
[matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/9577662/matrixDiff.txt)

I have a few more diffs, but they are not unexpected.

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Pull request merged**